packages/nextjs/hooks/flashbotRecoveryBundle/useGasEstimation.ts

```diff
@@ -20,51 +20,90 @@ export const useGasEstimation = () => {
     }),
   );
   const { showError } = useShowError();
+
+  // Cache for gas estimates and bundle hash
+  const lastBundleHash = useRef<string>("");
+  const lastEstimates = useRef<BigNumber[]>([]);
+  const lastBaseFeeTimestamp = useRef<number>(0);
+  const lastBaseFee = useRef<BigNumber>(BigNumber.from(0));
+
+  // Function to generate a hash for the transaction bundle
+  const getBundleHash = useCallback((txs: RecoveryTx[]): string => {
+    return txs.map(tx => tx.toEstimate?.data || "").join("");
+  }, []);
+
+  const maxBaseFeeInFutureBlock = async () => {
+    const now = Date.now();
+    // Cache base fee for 1 block (~12 seconds)
+    if (now - lastBaseFeeTimestamp.current < 12000 && !lastBaseFee.current.isZero()) {
```

Comment:
Nice. This is a bit of a tradeoff because potentially we will trail the current block by 11.99 seconds but I don't think it matters that much. Worth the call savings.